visuals/two_nights.py

```diff
@@ -0,0 +1,330 @@
+from datetime import datetime, time
+
+import astropy.units as astro_units
+import julian
+import pandas as pd
+import plotly.graph_objs as go
+import plotly.offline as pyo
+import requests
+from astroplan import Observer
+from astropy.time import Time
+
+from abstract_visualization import Visualization
+
+
+class TwoNights(Visualization):
+    """
+    This file contains the functions for creating a graph that overlays the
+    light frequency and temperature data from a sensor from two different
+    nights.
+    """
+
+    def __init__(
+        self,
+        evening_file1,
+        morning_file1,
+        evening_file2,
+        morning_file2,
+        sensor_latitude,
+        sensor_longitude,
+        sensor_number,
+    ):
+        self.evening_file1 = evening_file1
+        self.morning_file1 = morning_file1
+
+        self.evening_file2 = evening_file2
+        self.morning_file2 = morning_file2
+
+        self.sensor_latitude = sensor_latitude
+        self.sensor_longitude = sensor_longitude
+        self.sensor_location = Observer(
+            longitude=self.sensor_longitude * astro_units.deg,
+            latitude=self.sensor_latitude * astro_units.deg,
+            elevation=879 * astro_units.m,
+        )
+
+        self.sensor_number = sensor_number
+
+        self.m1, self.e1, self.m2, self.e2 = self._import_files()
+
+        self._parse_data(self.m1)
+        self._parse_data(self.e1)
+
+        self._parse_data(self.m2)
+        self._parse_data(self.e2)
+
+        self.evening_day1 = self.e1.loc[0, "Date"]
+        self.morning_day1 = self.m1.loc[0, "Date"]
+
+        self.evening_day2 = self.e2.loc[0, "Date"]
+        self.morning_day2 = self.m2.loc[0, "Date"]
+
+    def _import_files(self):
+        """
+        This method pulls the relevant sensor data file for the given night
+        and the following morning from Google Drive.
+        """
+        folder = "../streamlit/files/"
+        file_path_morn1 = folder + self.morning_file1
+        file_path_eve1 = folder + self.evening_file1
+
+        file_path_morn2 = folder + self.morning_file2
```

Comment:
Looks like you have some duplicated code over the next 17 lines. Save this functionality to an outside function.